src/classes/providers/YouTube.ts

```diff
@@ -150,8 +141,194 @@ export default class YouTube extends VideoProvider {
+    if ([EmbedType.Undefined, EmbedType.Video].indexOf(this.embedType) < 0) {
+      // don't reset type
+      return;
+    }
+    if (typeof params.get("playlist") === "undefined") {
+      return;
+    }
+    this.setStartVideo();
+    this.embedType = EmbedType.Playlist;
+    this.id = params.get("playlist") as string;
+  }
+
+  /**
+   * Fetch a video from a source URL
+   *
+   * @param path  The path to search for the video ID in
+   * @param params  The params to search for the video ID in
+   */
+  private fetchVideoFromSource(path: string, params: ParameterMap): void {
+    if (this.embedType !== EmbedType.Undefined) {
+      // don't reset type
+      return;
+    }
+    const match = path.match(new RegExp("/([^/]+?)$"));
+    if (match) {
+      this.embedType = EmbedType.Video;
+      [, this.id] = match;
+      return;
+    }
+    if (typeof params.get("v") !== "undefined") {
+      this.embedType = EmbedType.Video;
+      this.id = params.get("v") as string;
+    }
+  }
+
+  /**
+   * Sometimes a playlist or list will be configured to start with a specific
+   * video
+   *
+   * Converts a stored ID to a start video ID.
+   */
+  private setStartVideo(): void {
+    if (this.embedType !== EmbedType.Video) {
+      return;
+    }
+    this.embedType = EmbedType.Undefined;
+    this.startVideoID = this.id;
+  }
+
+  /**
+   * Grabs the timetamp from the given parameters if it's found,
+   * and saves it on the object.
+   *
+   * @param params  The parameters to parse
+   */
+  private fetchTimestampFromParams(params: ParameterMap): void {
+    if (
+      typeof params.get("start") !== "undefined" &&
+      parseInt(params.get("start") as string, 10) > 0
+    ) {
+      this.timestamp = new VideoTimestamp(params.get("start") as string);
+    } else if (params.get("t")) {
+      this.timestamp = new VideoTimestamp(params.get("t"));
+    }
+  }
+
+  /**
+   * Parse a source input that isn't a URL
+   *
+   * @param source  The source to parse
+   */
+  private parseNonURL(source: string): void {
+    if (this.embedType === EmbedType.Undefined) {
+      // don't reset type
+      return;
+    }
+    if (source.match(/^[a-zA-Z0-9_-]{11}$/)) {
+      // With no URL to go off of maybe it's a video ID?
+      // YouTube video IDs may change but this matches the current format
+      // https://stackoverflow.com/a/4084332
+
+      this.embedType = EmbedType.Video;
+      this.id = source;
+    }
+  }
+
+  /**
+   * Generate a list embed URL
+   *
+   * @return  The generated URL or empty string if it could not be generated
+   */
+  private generateListEmbedURL(): string {
+    if (this.embedType !== EmbedType.List) {
+      return "";
+    }
+    const params = new ParameterMap();
+
+    params.set("list", this.id);
+    params.set("listtype", this.listType);
+
+    if (this.timestamp) {
+      params.set("start", this.timestamp.getSeconds().toString());
+    }
+
+    let sourceAddress = `https://www.youtube-nocookie.com/embed`;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cdayjr/video-tag/pullRequest?prid=7551208)
```suggestion
    let sourceAddress = "https://www.youtube-nocookie.com/embed";
```